### PR TITLE
PUBLISH 2/15 (TBD): Update docs for log data partition UI

### PR DESCRIPTION
**Pull request by eknuth**
Update the existing docs for managing data partitions in the Logs UI. Replaces an existing configuration field and updates screenshots.

Changes are confined to this [page](http://localhost:8000/docs/logs/ui-data/data-partitions/).

### CODE REVIEW

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-data-partition-rules-tutorial.mdx [hunk 1]
```diff
@@ -64,7 +63,19 @@ Available data partition rule fields include:
       </td>
 
       <td>
-        The matching criteria for this data partition rule. Once the rule is enabled, logs matching this criteria will be routed to the specified data partition.
+         <i>(deprecated)</i> The matching criteria for this data partition rule. Once the rule is enabled, logs matching this criteria will be routed to the specified data partition.
```
Comment (akristen) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-data-partition-rules-tutorial.mdx:66]: Is this recently deprecated? Any reason to keep the row?

**Review by akristen — CHANGES_REQUESTED** [review]
Made some changes + had a question about the deprecated row. Will approve once I hear back from ya! :)

**Comment by eknuth** [comment]
Thanks for checking it out!

This is the first notice of deprecation. We're going to keep the field around for a bit to allow people to migrate away. 

I'm happy to make that more clear if it would be helpful.

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-data-partition-rules-tutorial.mdx [hunk 2]
```diff
@@ -64,7 +63,19 @@ Available data partition rule fields include:
       </td>
 
       <td>
-        The matching criteria for this data partition rule. Once the rule is enabled, logs matching this criteria will be routed to the specified data partition.
+         <i>(deprecated)</i> The matching criteria for this data partition rule. Once the rule is enabled, logs matching this criteria will be routed to the specified data partition. 
+        This field has been deprecated and will eventually be replaced with the `nrql` field to allow for more flexibility when creating data partitions. If the rule is
+        specified with `nrql` that cannot be parsed into `matchingCriteria`, this field will be null.
```
Comment (rkbodenner) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-data-partition-rules-tutorial.mdx:68]: For this:
> specified with `nrql` that cannot be parsed
How about this instead?
> specified with a NRQL `WHERE` clause in the `nrql` field that cannot be parsed

File: src/content/docs/apis/nerdgraph/examples/nerdgraph-data-partition-rules-tutorial.mdx [hunk 3]
```diff
@@ -64,7 +63,19 @@ Available data partition rule fields include:
...
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        NRQL
+      </td>
+
+      <td>
+        The matching criteria for this data partition rule specified using a NRQL `where` clause. Once the rule is enabled, logs matching this criteria will be routed to the specified data partition.
```
Comment (rkbodenner) [inline src/content/docs/apis/nerdgraph/examples/nerdgraph-data-partition-rules-tutorial.mdx:78]: Maybe uppercase that WHERE to distinguish it from the other things we're putting in sans-serif font?

Also consider linking to the NRQL docs: https://docs.newrelic.com/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions/

**Comment by eknuth** [comment]
@rkbodenner updated

**Review by homelessbirds — DISMISSED** [review]

**Review by zuluecho9 — APPROVED** [review]

**Review by akristen — APPROVED** [review]

**Pull request merged**